DotNetDBF/DBFReader.cs

```diff
@@ -245,175 +244,156 @@ internal Object[] NextRecord(IEnumerable<int> selectIndexes, IList<int> sortedIn
+                                        int.TryParse(monthString, out month) &&
+                                        int.TryParse(dayString, out day))
+                                    {
+                                        recordObjects[i] = new DateTime(year, month, day);
+                                    }
+                                    else
+                                    {
+                                        recordObjects[i] = null;
+                                    }
                                 }
-                                else
+                                catch (ArgumentOutOfRangeException)
                                 {
+                                    /* this field may be empty or may have improper value set */
                                     recordObjects[i] = null;
                                 }
-                            }
-                            catch (ArgumentOutOfRangeException)
-                            {
-                                /* this field may be empty or may have improper value set */
-                                recordObjects[i] = null;
-                            }
-
-                            break;
 
+                                break;
+                            }
                         case NativeDbType.Float:
-
-                            try
                             {
-                                byte[] t_float = new byte[
-                                    _header.FieldArray[i].FieldLength
-                                    ];
-                                _dataInputStream.Read(t_float, 0, t_float.Length);
-                                String tParsed = CharEncoding.GetString(t_float);
-                                var tLast = tParsed.Substring(tParsed.Length - 1);
-                                if (tParsed.Length > 0
-                                    && tLast != " "
-                                    && tLast != DBFFieldType.Unknown)
+                                try
                                 {
-                                    recordObjects[i] = Double.Parse(tParsed, NumberStyles.Float | NumberStyles.AllowLeadingWhite);
+                                    byte[] t_float = new byte[_header.FieldArray[i].FieldLength];
+                                    _dataInputStream.Read(t_float, 0, t_float.Length);
+                                    string tParsed = CharEncoding.GetString(t_float);
+                                    string tLast = tParsed.Substring(tParsed.Length - 1);
+                                    if (tParsed.Length > 0
+                                        && tLast != " "
+                                        && tLast != DBFFieldType.Unknown)
+                                    {
+                                        recordObjects[i] = Convert.ToSingle(tParsed, NumberFormatProvider);

```

Comment:
Okay, I see what you are trying to point out. I agree with you, and noticed that you already fixed in recent pull request, along with a lot of another updates. So it seems I should close my pull request?
About real numbers precision - I just wanted to fix the thing when integer numbers were parsed like doubles. But still, I thing that not all types in DBF parsed to the correct type in C#. Yes, DBF has a different definition for decimal, double and float types than C#, however, there are different types of DBF that store this numbers in a different way. So I can't say for sure how to correctly convert DBF values to C#.